packages/ssx-sdk/src/ssx.ts

```diff
@@ -107,15 +113,93 @@ export class SSX {
+        ? true
+        : config.modules.encryption;
+    if (encryptionConfig !== false) {
+      if (typeof encryptionConfig === 'object') {
+        // determine which encryption module to use
+        // initialize encryption with the provided config
+        if (encryptionConfig.module === 'SignatureEncryption') {
+          this.encryption = new SignatureEncryption(
+            encryptionConfig,
+            this.userAuthorization
+          );
+        } else {
+          throw new Error(
+            `Encryption module ${encryptionConfig.module} not supported`
+          );
+        }
+        // } else if (typeof encryptionConfig === 'function') {
+        //   // Initialize encryption with the return from the provided function
+        //   this.encryption = encryptionConfig();
+      } else {
+        // encryption == true or undefined
+        // Initialize encryption with default config when no other condition is met
+        this.encryption = new SignatureEncryption({}, this.userAuthorization);
+      }
+      // IEncryption does not yet extend ISSXExtension
+      // this.extend(this.encryption);
+    }
+
+    // initialize storage module
+    // assume storage module is **enabled** if config.storage is not defined
+    const storageConfig =
+      config?.modules?.storage === undefined ? true : config.modules.storage;
+    if (storageConfig !== false) {
+      if (typeof storageConfig === 'object') {
+        // Initialize storage with the provided config
+        this.storage = new KeplerStorage(storageConfig, this.userAuthorization);
+        // } else if (typeof storageConfig === 'function') {
+        //   // Initialize storage with the return from the provided function
+        //   this.storage = config.modules.storage();
+      } else {
+        // storage == true or undefined
+        // Initialize storage with default config when no other condition is met
+        this.storage = new KeplerStorage(
+          { prefix: 'ssx' },
+          this.userAuthorization
+        );
+      }
+      this.extend(this.storage);
+    }
+
+    // initialize data vault module
+    // assume data vault is **disabled** if config.dataVault is not defined
+    const dataVaultConfig =
+      config?.modules?.dataVault === undefined ? false : config.modules.dataVault;
+    if (dataVaultConfig !== false) {
+      if (typeof dataVaultConfig === 'object') {
+        // Initialize data vault with the provided config
+        this.dataVault = new KeplerDataVault(
+          dataVaultConfig,
+          this.userAuthorization,
+          this.encryption
+        );
+        // } else if (typeof dataVaultConfig === 'function') {
+        //   // Initialize data vault with the return from the provided function
+        //   this.dataVault = dataVaultConfig();
+      } else {
+        // dataVault == true or undefined
+        // Initialize data vault with default config when no other condition is met
+        this.dataVault = new KeplerDataVault(
```

Comment:
Style question: Currently the API is `ssx.dataVault.get("key")`. Should we move it to `ssx.datavault.get("key")`?